web/Controllers/LandlordController.cs

```diff
@@ -330,13 +323,45 @@ public async Task<ActionResult> Dashboard([FromRoute] int id)
         {
             return StatusCode(404);
         }
-
-        var viewModel = LandlordProfileModel.FromDbModel(landlord);
+        var landlordViewProperties = landlord.Properties.Select(PropertyViewModel.FromDbModel).OrderByDescending(p => p.PropertyId).Take(2).ToList();
+        var allPropertyDetails = new List<PropertyDetailsViewModel>();
+        foreach (var property in landlordViewProperties)
+        {
+            var fileNames = await _azureStorage.ListFileNames("property", property.PropertyId);
+            allPropertyDetails.Add(new PropertyDetailsViewModel
+            {
+                Images = GetFilesFromFileNames(fileNames, property.PropertyId),
+                Property = property
+            });
+        }
+        var viewModel = new LandlordDashboardViewModel
+        {
+            CurrentLandlord = LandlordProfileModel.FromDbModel(landlord),
+            Properties = allPropertyDetails,
+            PropertyTypeCount = _propertyService.CountProperties(id)
+        };
         return View("Dashboard", viewModel);
     }
 
-    [HttpGet]
-    [Route("me/dashboard")]
+    private List<ImageFileUrlModel> GetFilesFromFileNames(List<string> fileNames, int propertyId)
+    {
+        return fileNames.Select(fileName =>
+            {
+                var url = Url.Action("GetImage", new { propertyId, fileName })!;
+                return new ImageFileUrlModel(fileName, url);
+            })
+            .ToList();
+    }
+    
+    [HttpGet("{propertyId:int}/{fileName}")]
+    public async Task<IActionResult> GetImage(int propertyId, string fileName)
+    {
+        var (data, fileType) = await _azureStorage.DownloadFile("property", propertyId, fileName);

```

Comment:
Have you considered that the filename may get url encoded?

Reply:
This is one of Alice's old methods from the property controller that I have basically just copied over for the time being so we can get images working. There's a ticket I've created (BNH-167) to essentially refactor the image getting process and put it inside a service, so once I get round to that ticket this should all change (I attempted to do this during this ticket but it was quite complicated and took a long time so I figured a new ticket would be best). This method seems to do the trick for now at least :)